recipes-security/optee/optee-os/0001-arm-generic-allow-platforms-to-reserve-devices-for-e.patch

```diff
@@ -0,0 +1,199 @@
+From: Jorge Ramirez-Ortiz <jorge@foundries.io>
+Date: Tue, 10 Dec 2019 21:27:12 +0100
+Subject: [PATCH] arm: generic: allow platforms to reserve devices for
+ exclusive access
+
+Drivers defined in the platform specific main.c module will be added
+to the dts and to the generated overlay (if overlay generation is
+requested)
+
+In order to do that, the platforms need to define an interface in
+their main.c module through which the generic code can retrieve the
+driver names.
+
+An example for such an interface can be seen below:
+
+const char *main_get_optee_exclusive_node_name(unsigned int i)
+{
+	static const char *exclusive_drivers[] = {
+		"/ahb-bridge0@40000000/wdog@403D0000",
+		NULL,
+	};
+
+	if (i < ARRAY_SIZE(exclusive_drivers))
+		return exclusive_drivers[i];
+
+	return NULL;
+}
+
+Signed-off-by: Jorge Ramirez-Ortiz <jorge@foundries.io>
+---
+ core/arch/arm/include/kernel/generic_boot.h |   2 +-
+ core/arch/arm/kernel/generic_boot.c         | 116 ++++++++++++++++++++
+ 2 files changed, 117 insertions(+), 1 deletion(-)
+
+diff --git a/core/arch/arm/include/kernel/generic_boot.h b/core/arch/arm/include/kernel/generic_boot.h
+index 146a37c8..2127764b 100644
+--- a/core/arch/arm/include/kernel/generic_boot.h
++++ b/core/arch/arm/include/kernel/generic_boot.h
+@@ -24,7 +24,7 @@ void generic_boot_init_secondary(unsigned long nsec_entry);
+ 
+ void main_init_gic(void);
+ void main_secondary_init_gic(void);
+-
++const char* main_get_optee_exclusive_node_name(unsigned int i);
+ void init_sec_mon(unsigned long nsec_entry);
+ void init_tee_runtime(void);
+ 
+diff --git a/core/arch/arm/kernel/generic_boot.c b/core/arch/arm/kernel/generic_boot.c
+index f1a5874a..66879b55 100644
+--- a/core/arch/arm/kernel/generic_boot.c
++++ b/core/arch/arm/kernel/generic_boot.c
+@@ -27,6 +27,7 @@
+ #include <sm/psci.h>
+ #include <sm/tee_mon.h>
+ #include <stdio.h>
++#include <string.h>
+ #include <trace.h>
+ #include <utee_defines.h>
+ #include <util.h>
+@@ -106,6 +107,12 @@ __weak void main_secondary_init_gic(void)
+ {
+ }
+ 
++__weak const char* main_get_optee_exclusive_node_name(unsigned int i)
++{
++	(void) i;
++	return NULL;
++}
++
+ #if defined(CFG_WITH_ARM_TRUSTED_FW)
+ void init_sec_mon(unsigned long nsec_entry __maybe_unused)
+ {
+@@ -701,6 +708,112 @@ static int dt_add_psci_node(struct dt_descriptor *dt)
+ 	return 0;
+ }
+ 
++static char *strtok(char *str, const char *delims)
++{
++	static char *pos = NULL;
++	char *start = NULL;
++
++	if (str)
++		pos = str;
++
++	if (pos) {
++		while (*pos && strchr(delims, *pos)) {
++                    pos++;
++		}
++
++		if (*pos) {
++			start = pos;
++			while (*pos && !strchr(delims, *pos)) {
++				pos++;
++			}
++
++			if (*pos)
++				*pos++ = '\0';
++		}
++	}
++
++	return start;
++}
++
++static int add_descriptor_node(struct dt_descriptor *dt, const char *driver)
++{
++	unsigned int i, j;
++	char root_name[50];
++	char buffer[100];
++	char *item[10];
```

Comment:
Care to add a comment on how you decided to set up the sizes for all these items?